frost-core/src/keys/refresh.rs

```diff
@@ -0,0 +1,106 @@
+//! Refresh Shares
+//!
+//! Implements the functionality to refresh a share. This requires the participation
+//! of all the remaining signers. This can be done using a Trusted Dealer or
+//! DKG (not yet implemented)
+
+use std::collections::BTreeMap;
+
+use crate::{
+    keys::{
+        generate_coefficients, generate_secret_shares, validate_num_of_signers,
+        CoefficientCommitment, PublicKeyPackage, SigningKey, SigningShare, VerifyingShare,
+    },
+    Ciphersuite, CryptoRng, Error, Field, Group, Identifier, RngCore, Scalar,
+};
+
+use super::{SecretShare, VerifiableSecretSharingCommitment};
+
+/// Refreshes shares using a trusted dealer
+pub fn calculate_zero_key<C: Ciphersuite, R: RngCore + CryptoRng>(
+    old_pub_key_package: PublicKeyPackage<C>,
+    max_signers: u16,
+    min_signers: u16,
+    identifiers: &[Identifier<C>],
+    rng: &mut R,
+) -> Result<(Vec<SecretShare<C>>, PublicKeyPackage<C>), Error<C>> {
+    // Validate inputs
+
+    if identifiers.len() != max_signers as usize {
+        return Err(Error::IncorrectNumberOfIdentifiers);
+    }
+
+    validate_num_of_signers(min_signers, max_signers)?;
+
+    // Build zero key shares
+
+    let zero_key = SigningKey {
+        scalar: <<C::Group as Group>::Field>::zero(),
+    };
+
+    let coefficients = generate_coefficients::<C, R>(min_signers as usize - 1, rng);
+
+    let zero_shares = generate_secret_shares(
+        &zero_key,
+        max_signers,
+        min_signers,
+        coefficients,
+        identifiers,
+    )?;
+
+    let mut verifying_shares: BTreeMap<Identifier<C>, VerifyingShare<C>> = BTreeMap::new();
+
+    for share in zero_shares.clone() {
+        let signer_public = SigningShare::into(share.signing_share);
+        verifying_shares.insert(share.identifier, signer_public);
+    }
+
+    let pub_key_package = PublicKeyPackage::<C> {
+        header: old_pub_key_package.header,
+        verifying_shares,
+        verifying_key: old_pub_key_package.verifying_key,
+    };
+
+    Ok((zero_shares, pub_key_package))
```

Comment:
As I mentioned in Slack, there will be an issue here: the first commitment will be the identity, and will fail to be serialized. We could remove it from the commitment vector and re-add it in `refresh_share`.

Also, at first, I thought that the right `verifying_key` to return it would be matching verifying key for the zero key. But that would cause the same issue. So we can keep using the old verifying key.

Additionally, most of this function could be replaced by calling `split()` since the logic is the same.